### PR TITLE
pkg/endpoint: check for regenerateBPF error earlier

### PR DESCRIPTION
**Pull request by aanm**
It's confusing to send an error to a function and not checking it
earlier closer to the function that returned that error. Made some
changes to make the code easier for readability.

This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5377)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: pkg/endpoint/endpoint.go
Comment (ianvernon) [inline pkg/endpoint/endpoint.go]: Please put this function in `directory.go`.

File: pkg/endpoint/policy.go
Comment (ianvernon) [inline pkg/endpoint/policy.go]: Update comment to reflect removal of error parameter.

File: pkg/endpoint/directory.go
Comment (ianvernon) [inline pkg/endpoint/directory.go:50]: Update documentation to remove references to regeneration error

**Review by ianvernon — CHANGES_REQUESTED** [review]

File: pkg/endpoint/directory.go
Comment (aanm) [reply pkg/endpoint/directory.go:50]: Done

File: pkg/endpoint/endpoint.go
Comment (aanm) [reply pkg/endpoint/endpoint.go]: Done

File: pkg/endpoint/policy.go
Comment (aanm) [reply pkg/endpoint/policy.go]: Done

**Comment by aanm** [comment]
test-me-please

File: pkg/endpoint/endpoint.go
Comment (ianvernon) [inline pkg/endpoint/endpoint.go]: I had `go fmt` do this in go 1.11, but the CI has a different version of `go fmt`, which causes the formatting precheck to fail. You may want to run `gofmt` of a specific version until we update the CI :/

**Review by ianvernon — APPROVED** [review]

File: pkg/endpoint/endpoint.go
Comment (ianvernon) [reply pkg/endpoint/endpoint.go]: Never mind, the build already got past that point. I stand corrected!

**Comment by aanm** [comment]
test-me-please

**Comment by tgraf** [comment]
@aanm Supportive of the changes, can we wait until #5358 is merged?

File: pkg/endpoint/directory.go
Comment (joestringer) [inline pkg/endpoint/directory.go:57]: I don't think that this line (the remove of any potentially existing failDir) will be necessary any more.

**Comment by raybejjani** [comment]
@aanm I'm afraid you lost a merge race somewhere. The PR now has conflicts, a terrible PR disease...

**Review by tgraf — CHANGES_REQUESTED** [review]
We will wait for #5358 to be merged

**Review by tgraf — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**
